core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -178,8 +178,28 @@ public Object updateEvent() {
 
   @Override
   protected void cleanUncommitted(Set<ManifestFile> committed) {
-    if (newManifest != null && !committed.contains(newManifest)) {
-      deleteFile(newManifest.path());
+    if (newManifests != null) {
+      // Delete newManifests that have not been committed and clear them from the list.
```

Comment:
This is used to explain the above comments about why keep the `committed` here. I just noticed the doc in the parent class. Removed it since there is already an elaborate doc.